task_1/sergii.garanenko/src/main/java/services/CommandService.java

```diff
@@ -0,0 +1,65 @@
+package services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CommandService {
+
+  @Autowired
+  private VersionControlService controlService;
+
+  public void listen() {
+    BufferedReader reader = new BufferedReader(new InputStreamReader(System.in, StandardCharsets.UTF_8));
+
+    while (true) {
+      try {
+        String line = reader.readLine();
+        if (line != null) {
+          if (line.startsWith("quit")) {
+            break;
+          }
+          List<String> commands = Arrays.stream(line.split(" "))
+              .filter(item -> item.length() != 0)
+              .collect(Collectors.toList());
+          if (commands.size() > 0) {
+            switch (commands.get(0)) {
```

Comment:
When you have lots of if or case statements, it is usually not good.
If another programmer needs to add another command (merge for example or anything), they will add another if or case and there is a chance that they'll break all your function by that fix. The best practice to use polymorphism instead of "ifs" or "cases", so every command will reside in its own class and if next developer will need to add something they will add a new class and your code will be safe.
That particular approach called "Open/Closed principle", I couldn't have found a good article to give you an example. However, I'm going to describe it in one of the following lessons.
If you want to further improve your code, I'd recommend reading the book: https://martinfowler.com/books/refactoring.html